examples/vision/adain.py

```diff
@@ -0,0 +1,675 @@
+loss (`Ls`). The `lambda` term is used to vary the amount of style
+transfered.
+
+![The total loss](https://i.imgur.com/Q5y1jUM.png)
+
+### Content Loss
+
+This is the Euclidean distance between the content image features
+and the features of the neural style transferred image.
+
+![The content loss](https://i.imgur.com/dZ0uD0N.png)
+
+Here the authors propose to use the output from the AdaIn layer `t` as
+the content target rather than using features of the original image as
+target. This is done to optimize faster convergence of the loss.
+
+### Style Loss
+
+Rather than using the more commonly used [Gram
+Matrix](https://mathworld.wolfram.com/GramMatrix.html) the authors
+propose to compute the difference between the statistical features
+(mean and variance) which makes it conceptually cleaner. This can
+easily visualized in the following equation
+
+![The style loss](https://i.imgur.com/Ctclhn3.png)
+
+where `theta` denotes the layers in VGG-19 used to compute the loss.
+In this case this corresponds to:
+
+- `block1_conv1`
+- `block1_conv2`
+- `block1_conv3`
+- `block1_conv4`
+
+"""
+
+
+def get_loss_net():
+    vgg19 = keras.applications.VGG19(
+        include_top=False, weights="imagenet", input_shape=(*IMAGE_SIZE, 3)
+    )
+    vgg19.trainable = False
+    layer_names = ["block1_conv1", "block2_conv1", "block3_conv1", "block4_conv1"]
+    outputs = [vgg19.get_layer(name).output for name in layer_names]
+    mini_vgg19 = keras.Model(vgg19.input, outputs)
+
+    inputs = layers.Input([*IMAGE_SIZE, 3])
+    mini_vgg19_out = mini_vgg19(inputs)
+    return keras.Model(inputs, mini_vgg19_out, name="loss_net")
+
+
+"""
+## Neural Style Transfer
+
+This is the trainer module. We wrap the encoder and decoder inside a
+trainer module so that we can harness the beautiful `model.fit` API. 
+"""
+
+
+class NST(tf.keras.Model):
+    def __init__(self, encoder, decoder, ada_in, loss_net, lamb):
+        super().__init__()
+        self.encoder = encoder
+        self.decoder = decoder
+        self.ada_in = ada_in
+        self.loss_net = loss_net
+        self.lamb = lamb
+
+    def compile(self, optimizer, loss_fn):
+        super().compile()
+        self.optimizer = optimizer
+        self.loss_fn = loss_fn
+        self.style_loss_tracker = keras.metrics.Mean(name="style_loss")
+        self.content_loss_tracker = keras.metrics.Mean(name="content_loss")
+        self.total_loss_tracker = keras.metrics.Mean(name="total_loss")
+
+    def train_step(self, inputs):
+        style, content = inputs
+
+        # Initialize the content and style loss.
+        loss_content = 0.0
+        loss_style = 0.0
+
+        with tf.GradientTape() as tape:
+            # Encode the style and content image.
+            style_enc = self.encoder(style)
+            content_enc = self.encoder(content)
+
+            # Compute the AdaIN feature map.
+            t = self.ada_in(style=style_enc, content=content_enc)
```

Comment:
anything more descriptive we can call `t`?  it's ok if not.

Reply:
The term ```t``` (target feature maps) is taken from the paper itself
![image](https://user-images.githubusercontent.com/44690292/141061422-b5eb32b2-bff3-4fcd-9300-3a2dc08416d2.png)
So we think its best to leave it as ```t```